src/omt/omt_optimizer.cpp

```diff
@@ -39,32 +41,167 @@ std::unique_ptr<OMTOptimizer> OMTOptimizer::getOptimizerForNode(TNode targetNode
-    SmtEngine* parentSMTSolver, bool needsTimeout, unsigned long timeout)
+Node OMTOptimizer::mkStrongIncrementalExpression(
+    NodeManager* nm, TNode lhs, TNode rhs, OptimizationObjective& objective)
+{
+  constexpr const char lhsTypeError[] =
+      "lhs type does not match or is not implicitly convertable to the target "
+      "type";
+  constexpr const char rhsTypeError[] =
+      "rhs type does not match or is not implicitly convertable to the target "
+      "type";
+  TypeNode targetType = objective.getTarget().getType();
+  switch (objective.getType())
+  {
+    case OptimizationObjective::MINIMIZE:
+    {
+      if (targetType.isInteger() || targetType.isReal())
+      {
+        Assert(lhs.getType().isInteger() || lhs.getType().isReal())
+            << lhsTypeError;
+        Assert(rhs.getType().isInteger() || rhs.getType().isReal())
+            << rhsTypeError;
+        return nm->mkNode(Kind::LT, lhs, rhs);
+      }
+      else if (targetType.isBitVector())
+      {
+        Assert(lhs.getType() == targetType) << lhsTypeError;
+        Assert(rhs.getType() == targetType) << rhsTypeError;
+        return (objective.bvIsSigned())
+                   ? (nm->mkNode(Kind::BITVECTOR_SLT, lhs, rhs))
+                   : (nm->mkNode(Kind::BITVECTOR_ULT, lhs, rhs));
+      }
+      else if (targetType.isFloatingPoint())
+      {
+        Assert(lhs.getType() == targetType) << lhsTypeError;
+        Assert(rhs.getType() == targetType) << rhsTypeError;
+        return nm->mkNode(Kind::FLOATINGPOINT_LT, lhs, rhs);
+      }
+      else
+      {
+        CVC5_FATAL() << "Target type does not support optimization";
+      }
+    }
+    case OptimizationObjective::MAXIMIZE:
+    {
+      if (targetType.isInteger() || targetType.isReal())
+      {
+        Assert(lhs.getType().isInteger() || lhs.getType().isReal())
+            << lhsTypeError;
+        Assert(rhs.getType().isInteger() || rhs.getType().isReal())
+            << rhsTypeError;
+        return nm->mkNode(Kind::GT, lhs, rhs);
+      }
+      else if (targetType.isBitVector())
+      {
+        Assert(lhs.getType() == targetType) << lhsTypeError;
+        Assert(rhs.getType() == targetType) << rhsTypeError;
+        return (objective.bvIsSigned())
+                   ? (nm->mkNode(Kind::BITVECTOR_SGT, lhs, rhs))
+                   : (nm->mkNode(Kind::BITVECTOR_UGT, lhs, rhs));
+      }
+      else if (targetType.isFloatingPoint())
+      {
+        Assert(lhs.getType() == targetType) << lhsTypeError;
+        Assert(rhs.getType() == targetType) << rhsTypeError;
+        return nm->mkNode(Kind::FLOATINGPOINT_GT, lhs, rhs);
+      }
+      else
+      {
+        CVC5_FATAL() << "Target type does not support optimization";
+      }
+    }
+      // there's no default case
+      // because if more cases in OptimizationObjective::ObjectiveType are added
+      // compiler will issue a warning here in compile time instead of pushing
+      // it to run time.
```

Comment:
Please make this assumption explicit by adding a default case with an assertion of false. Compiler warnings are sometimes missed, and it is better to be explicit with assertion regarding what you are assuming.

Reply:
Ok, but the default case is falling through to the end, which is an unreachable statement as desired.

Reply:
added